invest_processes/src/invest_processes/slurm_manager.py

```diff
@@ -0,0 +1,361 @@
+            while True:
+                # check the 'state' string from the job data in sacct
+                status = subprocess.run([
+                    'sacct', '--noheader', '-X',
+                    '-j', job_id,
+                    '-o', 'State'
+                ], capture_output=True, text=True, check=True).stdout.strip()
+                LOGGER.debug(f'Status of slurm job {job_id}: {status}')
+
+                if status == 'COMPLETED':
+                    break
+                time.sleep(1)
+
+            # get the exit code from the job data in sacct
+            exit_code = int(subprocess.run([
+                'sacct', '--noheader', '-X', '-j', job_id, '-o', 'ExitCode'
+            ], capture_output=True, text=True, check=True).stdout.strip().split(':')[0])
+            LOGGER.debug(f'Exit code of slurm job {job_id}: {exit_code}')
+
+            if exit_code != 0:
+                LOGGER.error(f'Job {job_id} finished with non-zero exit code: {exit_code}')
+
+            outputs = processor.process_output(workspace_dir)
+            outputs['workspace'] = workspace_dir
+
+            # TODO: copy slurm job workspace to public bucket
+            # LOGGER.debug(f'Copying workspace for job {job_id} to bucket')
+
+            if requested_response == RequestedResponse.document.value:
+                outputs = {
+                    'outputs': [outputs]
+                }
+            current_status = JobStatus.successful
+
+        except Exception as err:
+            # TODO assess correct exception type and description to help users
+            # NOTE, the /results endpoint should return the error HTTP status
+            # for jobs that failed, the specification says that failing jobs
+            # must still be able to be retrieved with their error message
+            # intact, and the correct HTTP error status at the /results
+            # endpoint, even if the /result endpoint correctly returns the
+            # failure information (i.e. what one might assume is a 200
+            # response).
+
+            current_status = JobStatus.failed
+            code = 'InvalidParameterValue'
+            outputs = {
+                'type': code,
+                'code': code,
+                'description': f'Error executing process: {err}'
+            }
+            LOGGER.exception(err)
+
+        return job_id, 'application/json', outputs, current_status
+
+    def submit_slurm_job(self, processor, data_dict):
+        """Submit a slurm job to execute the process.
+
+        Args:
+            processor (Processor): processor to be executed
+            data_dict (dict): user data to pass to the processor
+
+        Returns:
+            job_id, workspace_dir
+        """
+        # Create a workspace directory
+        workspace_root = os.path.abspath('workspaces')
+        workspace_dir = os.path.join(workspace_root, f'slurm_wksp_{time.time()}')
+        os.makedirs(workspace_dir)
```

Comment:
However unlikely, it seems possible that under heavy, parallelized loads we may end up with two workspaces being created at exactly the same time.  If you agree, a possible way to get avoid this condition entirely might be to use `tempfile.mkdtemp` ([docs](https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp)) which is guaranteed to avoid such race conditions.  So an approximate equivalent to the lines you have here with the same information might in the directory name might be

```python
workspace_dir = tempfile.mkdtemp(dir=workspace_root, prefix=f'slurm_wksp_{time.time()}_')
```

which would make a new directory within `workspace_root` named something like `slurm_wksp_1762386391.142703_mdchbkm5`.  Maybe this would be useful here?

Reply:
I think we also don't need to include the timestamp if it's no longer needed for uniqueness